front-end-tools/src/components/WriteComponent.tsx

```diff
@@ -0,0 +1,618 @@
+                        <Alert variant="info">
+                            <div>
+                                This checkbox autofilled the <code>smart contract name</code>, the{' '}
+                                <code>entry point name</code>, and the <code>receive parameter schema</code> from the
+                                smart contract index.
+                            </div>
+                            <br />
+                            <div>
+                                <b>Uncheck</b> this box, if you want to manually fill in a{' '}
+                                <code>smart contract name</code>, an <code>entry point name</code>, or a{' '}
+                                <code>receive parameter schema</code>.
+                            </div>
+                            <br />
+                            <div>
+                                <b>Uncheck</b> and <b>check</b> this box again, if you want to load a new smart contract
+                                index.
+                            </div>
+                        </Alert>
+                    </>
+                )}
+
+                <Form.Group className="mb-3 d-flex justify-content-center">
+                    <Form.Check
+                        type="checkbox"
+                        id="isPayable"
+                        label="Is Payable"
+                        {...form.register('isPayable')}
+                        onChange={async (e) => {
+                            const isPayableRegister = form.register('isPayable');
+
+                            isPayableRegister.onChange(e);
+
+                            form.setValue('cCDAmount', 0);
+                        }}
+                    />
+                </Form.Group>
+
+                {isPayable && (
+                    <>
+                        <div className="box">
+                            <Form.Group className="mb-3">
+                                <Form.Label>CCD amount (micro):</Form.Label>
+                                <Form.Control
+                                    defaultValue={0}
+                                    type="number"
+                                    min="0"
+                                    {...form.register('cCDAmount', { required: true })}
+                                />
+                                <Form.Text />
+                                {form.formState.errors.cCDAmount && (
+                                    <Alert variant="info"> cCDAmount is required </Alert>
+                                )}
+                            </Form.Group>
+                        </div>
+                        <br />
+                    </>
+                )}
+
+                <Form.Group className="mb-3 d-flex justify-content-center">
+                    <Form.Check
+                        type="checkbox"
+                        id="hasInputParameter"
+                        label="Has Input Parameter"
```

Comment:
When I click on the label here (not the checkbox) then the page scrolls to the top. That's odd.

(I discovered this by accident :))

Reply:
I think this is caused because some of the `id`s are duplicated. Meaning if you clicked on the field it jumped to the first checkbox with this id on the page (which was somewhere at the top). I removed duplicated `id`s.